### ISSUE TITLE
All jobs silently fail after upgrading from 1.8.0 -> 1.8.1+ with "work-horse terminated unexpectedly"

### ISSUE THREAD

**Issue by atainter** [rq/rq#1678]
I'm not sure what is causing this. Here is the warning in the console:

```
Moving job to FailedJobRegistry (work-horse terminated unexpectedly; waitpid returned 256)
```

Looks like the job doesn't make it to the failed job registry either because I can't fetch the job or print exc info. Any ideas?

**Comment by atainter** [comment]
After restarting, I was able to pull job exc_info. I also logged info about my local Redis instance which looks correct

```
Redis<ConnectionPool<Connection<host=localhost,port=6379,db=0>>>
exc_info: Work-horse was terminated unexpectedly (waitpid returned 256)

```

**Comment by selwin** [comment]
Hmm I'm not sure. Are you able to add logging statements somewhere around here to see what's going on? https://github.com/rq/rq/blob/master/rq/worker.py#L850

Is this only happening on a single server? 1.8.1 has been out for a while so this bug is unlikely to happen to a lot of people.

**Comment by kernelguardian** [comment]
Happens to me too rq version is 1.11.1

**Comment by selwin** [comment]
Are you also on MacOS @kernelguardian ?

**Comment by kernelguardian** [comment]
yes I am, PS it only happens if when i add a callback function

**Comment by kernelguardian** [comment]
I ran the test suite and it worked fine, however when I added the call back function it threw error,

`import_attribute
    attribute_owner = getattr(module, attribute_owner_name)
AttributeError: module '__main__' has no attribute ''`

**Comment by selwin** [comment]
Yeah probably, it's not an error I've seen before.

**Comment by Jeepend** [comment]
I meet with this issue too, with rq==1.11.1.
I finally solved it by splitting the callback functions into another module:

1. Create cbk.py, and move report_success and report_failure into it;
2. Import cbk in the main module, and call enqueue with on_success=cbk.report_success and on_failure=cbk.report_failure;
3. Then the callback functions will be executed in the background too.

It's a little weird, is it designed as this? I supported the callback functions should be executed in the main module.

**Comment by selwin** [comment]
I think the first thing to fix would be the error message which is not descriptive enough, it should provide context as to why work horse was stopped, which in this case is the failure in executing callbacks.

**Comment by Jeepend** [comment]
> I think the first thing to fix would be the error message which is not descriptive enough, it should provide context as to why work horse was stopped, which in this case is the failure in executing callbacks.

Exactly, and it would be better if adding some sample code like this in the [example](https://python-rq.org/docs/#job-callbacks):
`
from somewhere import report_success, report_failure

queue.enqueue(say_hello, on_success=report_success, on_failure=report_failure)
`
